main.tf

```diff
@@ -100,17 +132,18 @@ locals {
   #     }
   #   }
   # }
-
   _root_module_stack_configs = merge([for module, files in local._root_module_yaml_decoded : {
-    for file, content in files : "${module}-${trimsuffix(file, ".yaml")}" =>
+    for file, content in files :
+    local._multi_instance_structure ? "${module}-${trimsuffix(file, ".yaml")}" : module =>
     merge(
       {
         "project_root" = replace(format("%s/%s", var.root_modules_path, module), "../", "")
         "root_module"  = module,
```

Comment:
I can take this on or we can do it in another PR: The thing I would say about the `project_root` is that there are certain times where it's needed to customize the "project root". I put that in quotes because if specified, it only pulls the said directory in this case. e.g. the example only pulls the directory `examples/complete/components/random-pet`
![image](https://github.com/user-attachments/assets/9b69e6f6-60a8-4a97-93d4-9117708022dd)

But for certain tools like Terraspace, where the entire root of the whole repository is needed for base templates and configs and others, this `project_root` logic wouldn't work here since it only pulls that said directory at runtime.

Reply:
I would think to put it in a `try` with `content. project_root` to let it be passed in.

Reply:
@oycyc let's follow up with support for this 👍